universal-application-tool-0.0.1/app/services/question/QuestionServiceImpl.java

```diff
@@ -70,18 +73,21 @@ public QuestionDefinition update(QuestionDefinition definition) throws InvalidUp
                     .collect(ImmutableList.toImmutableList()));
   }
 
-  private boolean isValid(QuestionDefinition newDefinition) {
+  private Optional<QuestionServiceError> validate(QuestionDefinition newDefinition) {
```

Comment:
Question (blocking): how are validation errors for missing values surfaced to the user? What if, for example, the question is missing a name and also has an invalid path?

Suggestion: return an `ImmutableSet<QuestionServiceError>` and validate for presence as well

Reply:
Could this be an `ErrorAnd<Void, QuestionServiceError>`?

Reply:
Done.